### PR TITLE
fix(SonarCube): issues are resolved

### PR DESCRIPTION
**Pull request by mkholjuraev**
resolves: https://issues.redhat.com/browse/SPM-792

SonarCube reliability bugs are fixed. Maintainability bugs will be fixed in the next PR

### CODE REVIEW

File: src/Utilities/Helpers.js [hunk 1]
```diff
@@ -15,6 +15,10 @@ import { Flex, FlexItem, Tooltip } from '@patternfly/react-core';
 import messages from '../Messages';
 import { intl } from './IntlProvider';
 
+export const isSet = (variable) => (
```
Comment (jiridostal) [inline src/Utilities/Helpers.js:18]: Is there any reason why we need it?

File: src/SmartComponents/SystemDetail/SystemDetail.js [hunk 2]
```diff
@@ -8,12 +8,13 @@ import { intl } from '../../Utilities/IntlProvider';
...
 import { NoSystemData } from '../../PresentationalComponents/Snippets/NoSystemData';
 import propTypes from 'prop-types';
+import { isSet } from '../../Utilities/Helpers';
 
 const SystemDetail = ({ isInventoryApp }) => {
     const [activeTabKey, setActiveTabKey] = React.useState(0);
     const [areTabsHidden, setTabsHidden] = React.useState(false);
 
-    const entity = useSelector(({ entityDetails }) => entityDetails.entity && entityDetails.entity || {});
+    const entity = useSelector(({ entityDetails }) => isSet(entityDetails.entity) && entityDetails.entity || {});
```
Comment (jiridostal) [inline src/SmartComponents/SystemDetail/SystemDetail.js:17]:
```suggestion
    const entity = useSelector(({ entityDetails }) => entityDetails.entity || {});
```
Why so complicated? It should work the same

File: src/Utilities/Helpers.js [hunk 3]
```diff
@@ -330,8 +334,8 @@ export function sortCves(cves, index, direction) {
 
     const sortedCves = cves.sort(
         ({ cells: aCells }, { cells: bCells }) => {
-            const aCell = aCells[index].value && aCells[index].value || aCells[index].title;
-            const bCell = bCells[index].value && bCells[index].value || bCells[index].title;
+            const aCell = isSet(aCells[index].value) && aCells[index].value || aCells[index].title;
```
Comment (jiridostal) [inline src/Utilities/Helpers.js:337]:
```suggestion
            const aCell = aCells[index].value || aCells[index].title;
```
Also for other occurrences

**Review by jiridostal — CHANGES_REQUESTED** [review]
isSet function is not needed at all. Some resources:

- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_OR
- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_AND
- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

File: src/Utilities/Helpers.js (hunk 1 above)
Comment (mkholjuraev) [reply src/Utilities/Helpers.js:18]: I also think that this is not needed. But the issue itself is weird https://sonarqube.corp.redhat.com/project/issues?id=insights%3Apatchman-ui&issues=AXfUchTERQz83K5m01H6&open=AXfUchTERQz83K5m01H6

**Pull request merged**